vibi-dpu/src/pubsub/listener.rs

```diff
@@ -0,0 +1,121 @@
+    let msgtype_opt = attributes.get("msgtype");
+    if msgtype_opt.is_none() {
+        eprintln!("msgtype attribute not found in message : {:?}", attributes);
+    }
+    let msgtype = msgtype_opt.expect("Empty msgtype");
+    match msgtype.as_str() {
+        "install_callback" => {
+            println!("Processing install callback message");
+            let msg_data_res =  serde_json::from_slice::<InstallCallback>(data_bytes);
+            if msg_data_res.is_err() {
+                eprintln!("Error deserializing install callback: {:?}",
+                    msg_data_res.expect_err("No error in msg_data"));
+                    return;
+            }
+            let data = msg_data_res.expect("msg_data not found");
+            let code_async = data.installation_code.clone();
+            task::spawn(async move {
+                // handle_install_bitbucket(&code_async).await;
+                println!("Processed install callback message");
+            });
+        },
+        "webhook_callback" => {
+            task::spawn(async move {
+                // process_review(&data_bytes).await;
+                println!("Processed webhook callback message");
+            });
+        }
+        _ => {
+            eprintln!("Message type not found for message : {:?}", attributes);
+        }
+    };
+}
+
+pub async fn get_pubsub_client_config(keypath: &str) -> ClientConfig {
+    let credfile = CredentialsFile::new_from_file(keypath.to_string()).await
+        .expect("Failed to locate credentials file");
+    return ClientConfig::default()
+        .with_credentials(credfile)
+        .await
+        .expect("Unable to get PubSub Client config");
+}
+
+async fn setup_subscription(keypath: &str, topicname: &str) -> Subscription{
+    let config = get_pubsub_client_config(keypath).await;
+    let client = Client::new(config).await
+        .expect("Unable to create pubsub client to listen to messages");
+    let topic = client.topic(topicname);
+    let topic_res = topic.exists(None).await;
+    if topic_res.is_err() {
+        let e = topic_res.expect_err("No error found in topic_res");
+        if e.code() == Code::NotFound {
+            client.create_topic(topicname, None, None).await
+                .expect("Unable to create topic");
+        }
+        else {
+            eprintln!("Error getting topic: {:?}", e);
+        }
+    }
+    let subconfig = SubscriptionConfig {
+        enable_message_ordering: true,
+        ..Default::default()
+    };
+    let subscriptionname = format!("{topicname}-sub");
+    let subscription = client.subscription(&subscriptionname);
+    if !subscription.exists(None).await.expect("Unable to get subscription information") {
+        subscription.create(
+            topic.fully_qualified_name(), subconfig, None)
+            .await.expect("Unable to create subscription for listening to messages");
+    }
+    println!("sub = {:?}", &subscription);
+    subscription
+}
```

Comment:
Good one, I missed reconfiguring this function

Reply:
Will do it in another PR though, right now, we can assume that if we fail in getting the pubsub subcription, the application quits, which is fine since we can't do anything without the pubsub subscription anyways